ic-helpers/src/factory/state.rs

```diff
@@ -160,7 +160,19 @@ async fn consume_provided_cycles_or_icp(
     icp_fee: u64,
     controller: Principal,
 ) -> Result<u64, FactoryError> {
-    if ic_cdk::api::call::msg_cycles_available() > 0 {
+    if {
```

Comment:
This also can be rewritten with `ic-kit`

Reply:
So, we're not the first to stumble upon this problem :p